### ISSUE TITLE
The use of private attribute defined in base class not detected in derived class.

### ISSUE THREAD

**Issue by asincero** [PyCQA/pylint#2620]
### Steps to reproduce

```python
class Base:
    def __init__(self) -> None:
        self.__foobar = 'foobar'


class Derived(Base):
    def do_it(self) -> None:
        print(self.__foobar)


o = Derived()
o.do_it()
```
### Current behavior

pylint does not detect the use of "self.__foobar" in the do_it() method of Derived.  If we try to run this, Python will complain about "_Derived__foobar" not being defined (like it is supposed to).

### Expected behavior

pylint should flag the use of "self.__foobar" in the do_it() method of Derived as an error.

### pylint --version output

pylint 2.1.1
astroid 2.0.4
Python 3.7.0 (default, Sep 20 2018, 14:56:27) 
[GCC 6.3.0 20170516]

**Comment by PCManticore** [comment]
Thanks for the report @asincero !

**Comment by Pierre-Sassoulas** [comment]
Closing as this was resolved with the creation of ``unused-private-member``.

**Comment by clavedeluna** [comment]
Is the solution here still as suggested above, to create a new checker for `undefined-private-member`?

**Comment by Pierre-Sassoulas** [comment]
Not sure about the actual name. This is a runtime error so it should be a default check. Considering we already have ``no-member`` message it could probably be considered a false negative of ``no-member`` instead.
